lib/behaves.rb

```diff
@@ -6,8 +6,13 @@ def implements(*methods)
     @behaviors ||= Set.new(methods)
   end
 
+  def inject_behaviours (&block)
+    @injected_behaviours ||= block
+  end
+
   def behaves_like(klass)
     at_exit { check_for_unimplemented(klass) }
+    add_injected_behaviours(klass)
```

Comment:
Although `at_exit` ensures that the injection will run before the check, I still think it makes more sense to move `add_injected_behaviors(klass)` to above `at_exit`.

This is just so that the order of execution is still retained, and readers will be less confused. (for example, one unfamiliar with `at_exit` might think that we only inject behaviors after checking, instead of the other way around))

Reply:
Done!